app/uk/gov/hmrc/transitmovementsrouter/connectors/PushNotificationsConnector.scala

```diff
@@ -82,15 +82,20 @@ class PushNotificationsConnectorImpl @Inject() (httpClientV2: HttpClientV2)(impl
   private def pushNotificationMessageUpdate(movementId: MovementId, messageId: MessageId, notificationType: String): UrlPath =
     UrlPath.parse(s"$baseRoute/traders/movements/${movementId.value}/messages/${messageId.value}/$notificationType")
 
-  override def postMessageReceived(movementId: MovementId, messageId: MessageId, messageType: MessageType, sourceXml: Source[ByteString, ?])(implicit
+  override def postMessageReceived(
+    movementId: MovementId,
+    persistenceResponse: PersistenceResponse,
+    messageType: MessageType,
+    sourceXml: Source[ByteString, ?]
+  )(implicit
     hc: HeaderCarrier,
     ec: ExecutionContext
   ): EitherT[Future, PushNotificationError, Unit] =
     EitherT {
-      if (!appConfig.pushNotificationsEnabled) {
+      if (!appConfig.pushNotificationsEnabled || !persistenceResponse.shouldSendNotification) {
         Future.successful(Right(()))
       } else {
-        val request = createRequest(movementId, messageId, messageReceivedPath)
+        val request = createRequest(movementId, persistenceResponse.messageId, messageReceivedPath)
```

Comment:
we need to call this part based on persistenceResponse service boolean value. That part is missing and need to fix it.